test/cmd/lfstest-customadapter.go

```diff
@@ -0,0 +1,253 @@
+// +build testtools
+
+package main
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/github/git-lfs/api"
+	"github.com/github/git-lfs/httputil"
+	"github.com/github/git-lfs/progress"
+	"github.com/github/git-lfs/tools"
+)
+
+// This test custom adapter just acts as a bridge for uploads/downloads
+// in order to demonstrate & test the custom transfer adapter protocols
+// All we actually do is relay the requests back to the normal storage URLs
+// of our test server for simplicity, but this proves the principle
+func main() {
+
+	scanner := bufio.NewScanner(os.Stdin)
```

Comment:
Oh if you meant the writers rather than the reader, I just like the explicitness of `Flush()` and the flexibility to build up more complex messages in stages if I wanted to (I didn't need to but hey).